history/ChangeHistory.cs

```diff
@@ -104,11 +105,38 @@ public InteractionCheckpointOp() : base(true)
     /// on the Change nodes. To step forward/backward in chunks, Changes can be tagged with text strings.
     /// InteractionCheckpointOp is a special-case tagged node that you can use at the "end" of an interactive operation.
     /// </summary>
-    public class ChangeHistory
+    public class ChangeHistory : INotifyPropertyChanged
```

Comment:
I think yes, because this is the much more generic approach to the problem and this interface is exactly match to our needs. Implementing via this interface can help us implement "raising event when property changed " on all properties if it is needed.Though if you are insisting to change this i can change.

Reply:
This would imply that at the subscription site you need to check if the property that has changed is a particular property by using string literals. This is just a bit too much abstraction IMO.

Reply:
If this is not critical for you , can you do favour for me and merge this i will return to this and fix this as you want. Just now have a lot of bugs on me. If not no  problem i will change  this in the way you want.

Reply:
I would agree with Zelim.
This is not necessary to implement INotifyPropertyChanged interface. In client code we only need to know when the history is changed. This happens not so often, actually. Then, in client code, we can check ForwardStepsAvailable and BackStepsAvailable properties.
Regarding Sargis's argument, I would say that there are no other properties to be notified about in this class.
In order to save Sargis's time I can create my own PR that would solve the task.

Reply:
Check this PR please:
https://github.com/ZelimDamian/frame3Sharp/pull/4